pkg/fanal/analyzer/language/php/composer/composer.go

```diff
@@ -35,7 +36,15 @@ func (a composerLibraryAnalyzer) Analyze(_ context.Context, input analyzer.Analy
 
 func (a composerLibraryAnalyzer) Required(filePath string, _ os.FileInfo) bool {
 	fileName := filepath.Base(filePath)
-	return utils.StringInSlice(fileName, requiredFiles)
+	if !utils.StringInSlice(fileName, requiredFiles) {
+		return false
+	}
+
+	// we should skip `composer.lock` inside `vendor` folder
+	if sep := string(os.PathSeparator); strings.Contains(filePath, sep+"vendor"+sep) || strings.HasPrefix(filePath, "vendor"+sep) {
```

Comment:
I suppose the path separator could be `/` when parsing tar files even though it runs on Windows. I didn't test it though. `ToSlash` might be safer.

```suggestion
	for _, p := range strings.Split(filepath.ToSlash(filePath), "/") {
		if p == "vendor" {
			return false
		}
	}
```